src/widgets/dynamic-widgets/__tests__/dynamic-widgets-test.ts

```diff
@@ -401,7 +401,7 @@ describe('dynamicWidgets()', () => {
         dynamicWidgets({
           container: rootContainer,
           transformItems(_items, { results }) {
-            return results.userData[0].MOCK_facetOrder;
+            return results!.userData[0].MOCK_facetOrder;
```

Comment:
turning something non-optional to optional probably is a breaking change, especially as it isn't called with undefined results. Maybe allow the whole metadata parameter to be generic instead of only the results?